ports/webassembly/api.js

```diff
@@ -131,23 +131,31 @@ export async function loadMicroPython(options) {
         },
         pyimport: pyimport,
         runPython(code) {
+            const len = Module.lengthBytesUTF8(code);
+            const buf = Module._malloc(len + 1);
+            Module.stringToUTF8(code, buf, len + 1);
             const value = Module._malloc(3 * 4);
             Module.ccall(
                 "mp_js_do_exec",
                 "number",
-                ["string", "pointer"],
-                [code, value],
+                ["pointer", "number", "pointer"],
+                [buf, len, value],
             );
+            Module._free(buf);
             return proxy_convert_mp_to_js_obj_jsside_with_free(value);
         },
         runPythonAsync(code) {
+            const len = Module.lengthBytesUTF8(code);
```

Comment:
if `len` always needs a `+ 1` (for null code `\x00` I suppose) why isn't this just `const len = Module.lengthBytesUTF8(code) + 1;` instead? Not a blocker, just a silly question!

Reply:
Bravo

Reply:
`Module.stringToUTF8` will always add a null byte at the end, so there must be enough room for it.

But, below in the call to `mp_js_do_exec_async`, we pass just `len`.